doc/api/async_hooks.md

```diff
@@ -0,0 +1,275 @@
+}
+
+
+function init(uid, type, triggerId, handle) {
+  const activity = { uid,  type, parentUid }
+  stamp(activity, 'init')
+  activities.set(uid, activity)
+}
+
+
+function before(uid) {
+  stamp(activities.get(uid), 'before')
+}
+
+
+function after(uid) {
+  stamp(activities.get(uid), 'after')
+}
+
+
+function destroy(uid) {
+  stamp(activities.get(uid), 'destroy')
+}
+
+
+// Register the above functions into the life time events of async operations
+const asyncHook = async_hooks.createHook({ init, before, after, destroy })
+
+
+// Start listening on the above hooks
+asyncHook.enable()
+
+
+// Create an async operation and run it twice
+let count = 0
+const iv = setInterval(function ontimeout() {
+  if (++count === 2) clearInterval(iv)
+}, 100)
+
+
+// Wait for process to exit and then print the data about async operations we collected
+process.on('exit', onexit)
+function onexit() {
+  // Stop listening on the above hooks
+  asyncHook.disable()
+  for (let v of activities.values()) {
+    console.log(v)
+  }
+}
+```
+
+
+### Output
+
+
+Here we can see that a Timeout and a TIMERWRAP were created and when their lifetime events occurred.
+
+
+```
+{ uid: 2,
+  type: 'Timeout',
+  parentUid: 1,
+  init: [ '1.13 ms' ],
+  before: [ '107.72 ms', '209.41 ms' ],
+  after: [ '108.47 ms', '210.14 ms' ],
+  destroy: [ '209.91 ms', '210.17 ms' ] }
+{ uid: 3,
+  type: 'TIMERWRAP',
+  parentUid: 1,
+  init: [ '1.72 ms' ],
+  before: [ '107.31 ms', '209.23 ms' ],
+  after: [ '108.54 ms', '210.18 ms' ],
+  destroy: [ '210.20 ms' ] }
+```
+
+
+We will refer back to this example in the remaining documentation.
+
+
+## API
+
+
+### async_hooks.createHook()
+
+
+Registers functions to be called for different lifetime events of each async operation. The hooks are as follows and are
+all optional. Returns an `AsyncHook`
+
+
+* `init` {function}
+  * called _exactly_ once whenever an async operation is created
+  * arguments
+    * `uid` {number} a unique id for the async operation
+    * `type` {String} the type of the async operations, the types and their meanings are explained futher below
+      TODO(thlorenz)
+    * `triggerId` {number} the unique id of the async operation in which context this async operation was created
+    * `handle` {Object} reference to the handle representing the async operation, needs to be released during _destroy_
+* `before` {function}
+  * called whenever a callback is invoked from the context of this async operation
+  * arguments
+    * `uid` {number} a unique id for the async operation
+* `after` {function}
+  * called whenever a callback that was invoked from the context of this async operation finished executing
+  * arguments
+    * `uid` {number} a unique id for the async operation
+* `destroy` {function}
+  * called _exactly_ once whenever an async operation is destroyed
+  * arguments
+    * `uid` {number} a unique id for the async operation
+
+
+### AsyncHook.enable
```

Comment:
This should either be `asyncHook.enable()` or `AsyncHook#enable()`. Like this it looks like a static member (i.e. not one on the prototype).